### PR TITLE
Dev 25575

### PR DESCRIPTION
**Pull request by loaiabdelhalim**

### CODE REVIEW

**Review by JustinHorn — APPROVED** [review]
looks good

**Review by JustinHorn — CHANGES_REQUESTED** [review]
Briefly looked at you commit names. The format that we agreed up on is: 
dev-ticket**;** feat: message
please rename those commits. ( If you don't know how to I can show you)

File: src/adapters/CKEditorAdapter.ts [hunk 1]
```diff
@@ -35,11 +41,21 @@ export class CKEditorAdapter extends AbstractRichtextEditorAdapter {
...
   }
 
   getContent() {
-    return this.getEditor().getData();
+    // for version 4
+    if(this.editorVersion == 4){
+      return this.getEditor().getData();
+    }
+    // for version 5
+    return document.querySelector(this.editorIdHashed)!.innerHTML;
```
Comment (abhijeetnarvekar) [inline src/adapters/CKEditorAdapter.ts:58]: Hi,
The purpose of the adapter is to use editor API to fetch content. If we simply use HTML dom operation to get content there is no need to adapter.

If you look a v4 implementation it gets CK4 editor instance and uses getData() API of the editor.

Getting content via DOM can already be done using contenEditableAdapter. https://github.com/acrolinx/sidebar-sdk-js/blob/main/src/adapters/ContentEditableAdapter.ts#L35

I think we should use CK5 API to get content from editor and not DOM innerHTML

**Review by abhijeetnarvekar — CHANGES_REQUESTED** [review]

**Review by abhijeetnarvekar — CHANGES_REQUESTED** [review]

File: src/adapters/CKEditorAdapter.ts [hunk 2]
```diff
@@ -20,22 +20,45 @@ import {HasEditorID, ContentExtractionResult} from "./AdapterInterface";
...
   }
 
   getEditorDocument(): Document {
-    return this.getEditor().document.$ as any;
+    // for version 4
+    if(this.editorVersion == 4){
+      return this.getEditor().document.$ as any;
+    }
+    // for version 5
+    return document.querySelector(this.editorIdHashed)!.ownerDocument;
```
Comment (abhijeetnarvekar) [inline src/adapters/CKEditorAdapter.ts:61]: You can directly use the editor instance to the containing document, so you don't have to execute same query selector again and could also avoid adding the bang operator (non-null assertion operator)

Try looking into Ck5 API:
GetDomRoot: https://ckeditor.com/docs/ckeditor5/latest/api/module_engine_view_view-View.html#function-getDomRoot

`instance.editing.view.getDomRoot()`

OR

GetEditableElement: https://ckeditor.com/docs/ckeditor5/latest/api/module_core_editor_editorui-EditorUI.html#function-getEditableElement

`instance.ui.getEditableElement()`

File: src/adapters/CKEditorAdapter.ts [hunk 3]
```diff
@@ -68,6 +97,10 @@ export class CKEditorAdapter extends AbstractRichtextEditorAdapter {
   }
 
   isInWysiwygMode() {
-    return this.getEditor().mode === 'wysiwyg';
+    if(this.editorVersion == 4){
+      return this.getEditor().mode === 'wysiwyg';
+    }
+    const editorElement = this.getEditorDocument().querySelector('.ck-editor .ck-source-editing-button')!;
```
Comment (abhijeetnarvekar) [inline src/adapters/CKEditorAdapter.ts:103]: Please avoid using css selectors on buttons where possible, as there is high chance that they may break in minor updates also they might get overridden by the CMS application where CK editor is embedded.

Try using CK5 API

in CK5 source view is available as native plugin.
You can use the API to check if its enabled and currently in use.

Source Editing Demo: https://ckeditor.com/docs/ckeditor5/latest/features/source-editing.html

Source Editing API: https://ckeditor.com/docs/ckeditor5/latest/api/module_source-editing_sourceediting-SourceEditing.html


```
if (instance.plugins.has("SourceEditing")) {
  const sep = instance.plugins.get("SourceEditing");
  return sep.isEnabled && sep.isSourceEditingMode;
}
```

The plugin id "SourceEditing"is taken from docs: https://ckeditor.com/docs/ckeditor5/latest/api/module_source-editing_sourceediting-SourceEditing.html

File: src/adapters/CKEditorAdapter.ts [hunk 4]
```diff
@@ -20,22 +20,45 @@ import {HasEditorID, ContentExtractionResult} from "./AdapterInterface";
...
+        throw new Error(`Can't find ckeditor with id '${this.editorId}'`);
+      }
+      return ckeditor;
+    }
+    // CKEditor Version 5 detected
+    let editorDomElement = document.querySelector(this.editorIdHashed)!;
+    const isInlineEditor = editorDomElement.classList.contains('ck-editor__editable');
+    if(!isInlineEditor){
+      // Classic editor detected
+      editorDomElement = editorDomElement.nextElementSibling!.querySelector('.ck-editor__editable')!;
```
Comment (abhijeetnarvekar) [inline src/adapters/CKEditorAdapter.ts:48]: Is the usage of `nextElementSibling` observation based or does it have a reason?

**Review by abhijeetnarvekar — CHANGES_REQUESTED** [review]

File: src/adapters/CKEditorAdapter.ts (hunk 4 above)
Comment (loaiabdelhalim) [reply src/adapters/CKEditorAdapter.ts:48]: It does have a reason. If a classic editor is in use, the next element is the editor.

File: src/adapters/CKEditorAdapter.ts [hunk 5]
```diff
@@ -20,22 +20,45 @@ import {HasEditorID, ContentExtractionResult} from "./AdapterInterface";
...
   constructor(conf: HasEditorID) {
     super(conf);
     this.editorId = conf.editorId;
+    if(typeof window["CKEDITOR"] == 'undefined'){
+      this.editorVersion = 5;
+    }
+    this.editorIdHashed = "#" + this.editorId;
   }
 
   getEditor(): CKEDITOR.editor {
```
Comment (abhijeetnarvekar) [inline src/adapters/CKEditorAdapter.ts:35]: `CKEDITOR.editor` is return type for version 4. It's not valid for v5

File: src/adapters/CKEditorAdapter.ts (hunk 2 above)
Comment (loaiabdelhalim) [reply src/adapters/CKEditorAdapter.ts:61]: `instance.editing.view.getDomRoot()` and `instance.ui.getEditableElement()` cannot be used to access the parent document.

**Review by abhijeetnarvekar — APPROVED** [review]

**Review by JustinHorn — APPROVED** [review]

**Pull request merged**